### DISCUSSION TITLE
CDSM extraction from DSM

### DISCUSSION THREAD

**Discussion by AlexKDo** [UMEP-dev/UMEP#502] [Q&A] [answered]
Hello. I would like to calculate the outdoor thermal comfort for a specific area in Athens, Greece. I have the DSM and DEM files, but not the CDSM. In the DSM file that I have, the trees and buildings are mixed. Is there a way that I can somehow create a CDSM file from that?  Thank you in advance

**Comment by biglimp** [comment, marked as answer]
Yes, but you need a vector polygon footprint layer of all buildings. If you dont have that you could probably get one from Open Street Map. Then you du the following:

1. Subtract your DEM from your DSM to get object above ground
2. Buffer your building footprint with e.g. 2 meters
3. Add an attribute with zero values to your buffered layer
4. Burn these buffered polygons with zero value into your raster created in 1 and voila, you now have a CDSM.

However, you also need to clean your original DSM, i.e. remove trees. This can be done by creating an “inverted building footprint” and then do the process above again. A final step includes adding your DEM since your DSM should be meters above sea level.

I recommend QGIS for doing all these steps and let us know if you are having problems.

**Comment by AlexKDo** [comment]
Hello.. Thank you for your answer. I did all the steps you told me and i have my CDSM.

I iam trying to do the opposite, but somehow it seems impossible. I tried many different ways. but step 1 for DSM cleaning, after i do all the steps doesnt change a bit. U thing it has to do with the attribute table?

**Reply by biglimp** [reply]
Could be. Also check your CRS of your inverted polgonlayer.

**Reply by AlexKDo** [reply]
CRS are correct. So in my inverted polygonlayer, should I add an attribute field with zero values again? In the first try I added an attribute field like this.

![Στιγμιότυπο οθόνης 2023-03-11 143038](https://user-images.githubusercontent.com/81244161/224484606-2550974a-63ba-4220-bb7b-11811f204dc5.png)

In the invered polygonlayer, i created a new polygon in the same area, without the buildings and I added 0 value only in the new polygon - the other ones in the dame filed remain NULL. I also used the landuse polygonlayer - clipped it with inverted buffer buildings shp- did the same procedure as your example but still nothing changed in the above ground tiff.. 
Attribute table for this one is here:

![landusesnip](https://user-images.githubusercontent.com/81244161/224484964-dbe56236-6d4d-4474-b1be-80088651dce0.png)

**Reply by AlexKDo** [reply]
I finally did it. I created a new project and did everything again and somehow it happened! Thank you

**Comment by AlexKDo** [comment]
Alo again.. So I created CDSM and separated trees from DSM and I now have 2 files.
CDSM looks likes this.. I buffered the buildings and then I tried to make it better using the DSM:
![image](https://user-images.githubusercontent.com/81244161/224553506-dff9f468-569e-4eca-adb4-9df16d118188.png)

But I am trying to run eg SVF and this is the result I get. 1 one only with DSM and 2 with DSM & CDSM. 
1.
![image](https://user-images.githubusercontent.com/81244161/224553608-da25c2db-4fb8-4867-b00a-e9c5605eecf3.png)

2.
![image](https://user-images.githubusercontent.com/81244161/224553625-5f150a8f-d926-4fbb-82ca-3f4e952c0684.png)

Also.. in my try to create Daily Shadow pattern, the only DSM output was ok but then again..
![image](https://user-images.githubusercontent.com/81244161/224553797-279ff931-0234-4a17-802a-3723184e448b.png)

You think it might be a solution to create a CDSM with Tree Generator, using my clean DSM file? I am experimenting with a small square with not that muvh vegetation around.

**Comment by AlexKDo** [comment]
sorry, CDSM without DSM background looks like this
![image](https://user-images.githubusercontent.com/81244161/224554013-fdade239-e632-4e61-be07-2aa2f4bb26cb.png)

**Comment by biglimp** [comment]
All pixels tjat are not trees in CDSM should be 0, not NoData or very small numbers which happens some time when you have a sperarate DEM and DSM.

**Reply by AlexKDo** [reply]
I did it that way.. You think when I added the new file+ DEM file something changed? Because everything is 0, except from trees but it doesnt look like this when I did SVF and Daily Shadow. I painted 0 with blue so that you can see
![image](https://user-images.githubusercontent.com/81244161/224560735-3d2f3944-a549-4c33-98f0-bfabdf168355.png)

**Reply by AlexKDo** [reply]
So apparently 0= no data value. How did this happen?
![image](https://user-images.githubusercontent.com/81244161/224561684-4c9acc19-1ac2-4363-9ddc-d3f6d64d5421.png)

**Comment by biglimp** [comment]
Attach your dsm and cdsm and I can have a look.

**Reply by AlexKDo** [reply]
I created a respiratory, I am really new here so if there is any problem tell me please. Thank you 
https://github.com/AlexKDo/project-files.git

**Comment by AlexKDo** [comment]
Problem solved. I didnt have to add my created CDSM to the DEM file. thank you for your help!
